src/main/java/dk/aau/p4/abaaja/Lib/Visitors/SymbolTableVisitor.java

```diff
@@ -124,6 +124,10 @@ public void visit(VarDecNode node) {
             typeDescriptor = typeCheckingVisitor.visit(node.get_varDecType());
             if (typeDescriptor != null) {
                 Symbol symbol = new Symbol(node.get_id(), typeDescriptor);
+
+                if (typeDescriptor instanceof MctlArrayTypeDescriptor) {
+                    symbol.set_isInstantiated(true);
+                }
```

Comment:
Should this check also include structs? I am thinking that someone might want to copy an empty struct to add different values to it, but then it fails because they need to add some data to it before it is initialized?

Reply:
We can do that if you want :)